includes/Core/Storage/User_Setting/Array_Setting.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Storage\User_Setting\List
+ *
+ * @package   Google\Site_Kit\Core\Storage\User_Setting
+ * @copyright 2023 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Storage\User_Setting;
+
+use Google\Site_Kit\Core\Storage\User_Setting;
+
+/**
+ * Base class for a single user setting of the array type.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+abstract class Array_Setting extends User_Setting {
```

Comment:
This seems generally useful for settings and doesn't have anything specific to user settings. How about we make this a trait? Then only the `sanitize_array_item` method would need to be abstract. I think then we would be able to use this for other kinds of settings, not just user settings.